app/services/user_profile.rb

```diff
@@ -0,0 +1,70 @@
+class UserProfile
+  attr_reader :user, :tab
+  include Rails.application.routes.url_helpers
+
+  def initialize(user_id, tab, publications)
+    @user = User.find_by(id: user_id)
```

Comment:
If you expect the user to always exist and want to raise an exception if they don't, use :
```suggestion
    @user = User.find(user_id)
```

Reply:
Fine, currently the nil value is being used to display the same elements for the profile view. Maybe it will be good if this is addressed in another PR, adding a new view for a nonexistent user displayed directly after an exception is raised.